### ISSUE TITLE
TypeError: Can't add property _instantiateReactComponent, object is not extensible

### ISSUE THREAD

**Issue by lenny0702** [walmartlabs/react-ssr-optimization#13]
Firstly, I think this framework is a great idea. Totally solved our concern of react's sever rendering. Great work!

My problem is : I run a react server with this framework on my local machine. I don't want change the NODE_ENV value. But I still need to test if this is functioning. So I change the original code of this line 
[https://github.com/walmartlabs/react-ssr-optimization/blob/master/lib/index.js#L47](https://github.com/walmartlabs/react-ssr-optimization/blob/master/lib/index.js#L47)
to `if (false && process.env.NODE_ENV !== "production") {:`

But I got this:
TypeError: Can't add property _instantiateReactComponent, object is not extensible

How come? Is it related to my react's version? My react's version is 15,

**Comment by maximenajim** [comment]
Hi @lenny0702, thanks for your feedback.  This library intercepts React's instantiateReactComponent module by using a require() hook to avoid forking React.  We could evaluate doing this in a slightly different way.  Looking at the React code there is an Object.preventExtensions() being done in the non-production environments which causes the "object is not extensible" error:
https://github.com/facebook/react/blob/43130597021cbf9332675f4ca53e4f49c7001ca1/src/renderers/shared/stack/reconciler/instantiateReactComponent.js#L142-L148

In short, this is not related to the react version per se but simply the hook we are using to extend react doesn't really work outside of NODE_ENV == "production"
